packages/react-router-dom/index.tsx

```diff
@@ -1067,9 +1081,11 @@ let savedScrollPositions: Record<string, number> = {};
 function useScrollRestoration({
   getKey,
   storageKey,
+  skip = false,
 }: {
   getKey?: GetScrollRestorationKeyFunction;
   storageKey?: string;
+  skip?: boolean;
```

Comment:
Still not sure if this is 100% required or the best solution but going to play with it a bit more, need a way for Remix to communicate hydration info to skip initial restoration (which Remix did from an inline script)